### PR TITLE
Add code coverage for combobox

### PR DESCRIPTION
**Pull request by Nora-Zhou01**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for ComboBox to test its methods: GetItemHeight, ResetText, SelectAll

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11565)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 1]
```diff
@@ -2411,6 +2411,103 @@ public void ComboBox_Select_Item_By_Key(Keys key, int expectedKeyPressesCount, i
...
+    public void ComboBox_GetItemHeight_Invoke_ReturnsExpected(DrawMode drawMode)
+    {
+        int index = 0;
+        int expected = 15;
+        using ComboBox control = new()
+        {
+            DrawMode = drawMode,
+            ItemHeight = expected
+        };
+        control.Items.Add("Item1");
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:2427]: Maybe create a factory method for creating the control and avoid repetition, like this:

    ```cs
    private ComboBox CreateComboBox(DrawMode drawMode, int itemHeight = 15)
    {
        ComboBox control = new()
        {
            DrawMode = drawMode,
            ItemHeight = itemHeight
        };

        control.Items.Add("Item1");
        control.Items.Add("Item2");
        control.Items.Add("Item3");
        control.CreateControl();

        return control;
    }
    ```

Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:2427]: Maybe create a factory method for creating the control and avoid repetition, like this:

    ```cs
    private ComboBox CreateComboBox(DrawMode drawMode, int itemHeight = 15)
    {
        ComboBox control = new()
        {
            DrawMode = drawMode,
            ItemHeight = itemHeight
        };

        control.Items.Add("Item1");
        control.Items.Add("Item2");
        control.Items.Add("Item3");
        control.CreateControl();

        return control;
    }
    ```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 2]
```diff
@@ -2411,6 +2411,111 @@ public void ComboBox_Select_Item_By_Key(Keys key, int expectedKeyPressesCount, i
...
+
+    [WinFormsTheory]
+    [InlineData(-1)]
+    [InlineData(0)]
+    [InlineData(2)]
+    [InlineData(3)]
+    [InlineData(4)]
+    public void ComboBox_GetItemHeight_Index_ThrowsArgumentOutOfRangeException(int index)
+    {
+        using ComboBox control = new();
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:2450]: The `CreateComboBox()` factory method could be used here as well, like this:

```cs
using ComboBox control = CreateComboBox(DrawMode.OwnerDrawVariable);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 3]
```diff
@@ -2411,6 +2411,111 @@ public void ComboBox_Select_Item_By_Key(Keys key, int expectedKeyPressesCount, i
...
+    [InlineData(-1)]
+    [InlineData(0)]
+    [InlineData(2)]
+    [InlineData(3)]
+    [InlineData(4)]
+    public void ComboBox_GetItemHeight_Index_ThrowsArgumentOutOfRangeException(int index)
+    {
+        using ComboBox control = CreateComboBox(DrawMode.OwnerDrawVariable);
+        control.CreateControl(); // Ensure the handle is created
+        control.Items.Add("Item1");
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:2452]: The `CreateComboBox(DrawMode.OwnerDrawVariable)` method itself creates these 3 items, so adding them bellow again seem unecessary.

**Comment by LeafShi1** [comment]
@lonitra  Please help review this PR

**Review by lonitra — APPROVED** [review]
👍

**Pull request merged**